modules/rostests/apitests/ntdll/RtlCriticalSection.c

```diff
@@ -0,0 +1,374 @@
+        {
+            Status = pfnRtlInitializeCriticalSectionEx(&CritSect, 0, RTL_CRITICAL_SECTION_FLAG_FORCE_DEBUG_INFO);
+            ok_ntstatus(Status, STATUS_SUCCESS);
+            ok(CritSect.DebugInfo != NULL, "DebugInfo is %p\n", CritSect.DebugInfo);
+            ok(CritSect.DebugInfo != LongToPtr(-1), "DebugInfo is %p\n", CritSect.DebugInfo);
+            if ((CritSect.DebugInfo != NULL) && (CritSect.DebugInfo != LongToPtr(-1)))
+            {
+                ok_int(CritSect.DebugInfo->Type, 0);
+                ok_int(CritSect.DebugInfo->CreatorBackTraceIndex, 0);
+                ok_int(CritSect.DebugInfo->CreatorBackTraceIndexHigh, 0);
+                ok_ptr(CritSect.DebugInfo->CriticalSection, &CritSect);
+                ok(CritSect.DebugInfo->ProcessLocksList.Flink != NULL, "Flink is NULL\n");
+                ok(CritSect.DebugInfo->ProcessLocksList.Blink != NULL, "Blink is NULL\n");
+                if ((CritSect.DebugInfo->ProcessLocksList.Flink != NULL) &&
+                    (CritSect.DebugInfo->ProcessLocksList.Blink != NULL))
+                {
+                    ok_ptr(CritSect.DebugInfo->ProcessLocksList.Flink->Blink,
+                        &CritSect.DebugInfo->ProcessLocksList);
+                    ok_ptr(CritSect.DebugInfo->ProcessLocksList.Blink->Flink,
+                        &CritSect.DebugInfo->ProcessLocksList);
+                }
+                ok_long(CritSect.DebugInfo->EntryCount, 0);
+                ok_long(CritSect.DebugInfo->ContentionCount, 0);
+                ok_long(CritSect.DebugInfo->Flags, 0);
+                ok_int(CritSect.DebugInfo->SpareWORD, 0);
+            }
+        }
+    }
+    else
+    {
+        skip("RtlInitializeCriticalSectionEx not available.\n");
+    }
+}
+
+static
+DWORD
+WINAPI
+ThreadProc1(
+    _In_ LPVOID lpParameter)
+{
+    printf("ThreadProc1 starting\n");
+    RtlEnterCriticalSection(&CritSect);
+
+    SetEvent(hEventThread1Ready);
+
+    printf("ThreadProc1 waiting\n");
+    WaitForSingleObject(hEventThread1Cont, INFINITE);
+    printf("ThreadProc1 returned from wait\n");
+
+    RtlLeaveCriticalSection(&CritSect);
+
+    return 0;
+}
+
+static
+DWORD
+WINAPI
+ThreadProc2(
+    _In_ LPVOID lpParameter)
+{
+    printf("ThreadProc2 starting\n");
+    RtlEnterCriticalSection(&CritSect);
+
+    SetEvent(hEventThread2Ready);
+
+    printf("ThreadProc2 waiting\n");
+    WaitForSingleObject(hEventThread2Cont, INFINITE);
+    printf("ThreadProc2 returned from wait\n");
+
+    RtlLeaveCriticalSection(&CritSect);
+
+    return 0;
+}
+
+static
+void
+Test_Acquire(void)
+{
+    DWORD dwThreadId1, dwThreadId2;
+
+    RtlInitializeCriticalSection(&CritSect);
+
+    // Acquire once
+    RtlEnterCriticalSection(&CritSect);
+    ok_long(CritSect.LockCount, -2);
+    ok_long(CritSect.RecursionCount, 1);
+    ok_ptr(CritSect.OwningThread, UlongToHandle(GetCurrentThreadId()));
+    ok_ptr(CritSect.LockSemaphore, NULL);
+    //ok_size_t(CritSect.SpinCount, 0x20007d0);
```

Comment:
Might want to comment as to why this is commented out.

If the reason is that it depends on UP vs MP -- maybe just save the initial value and then compare with that to show "acquire doesn't change spin count"?